src/main/java/sparksoniq/jsoniq/runtime/iterator/primary/VariableReferenceIterator.java

```diff
@@ -31,68 +35,87 @@
 import java.util.Map;
 import java.util.TreeMap;
 
-import org.rumbledb.api.Item;
-
-public class VariableReferenceIterator extends LocalRuntimeIterator {
+public class VariableReferenceIterator extends HybridRuntimeIterator {
 
 
     private static final long serialVersionUID = 1L;
-    private SequenceType sequence;
+    private SequenceType _sequence;
     private String _variableName;
-    private List<Item> items = null;
-    private int currentIndex = 0;
+    private List<Item> _items = null;
+    private int _currentIndex = 0;
 
     public VariableReferenceIterator(String variableName, SequenceType seq, IteratorMetadata iteratorMetadata) {
         super(null, iteratorMetadata);
-        this._variableName = "$" + variableName;
-        this.sequence = seq;
+        _variableName = "$" + variableName;
+        _sequence = seq;
+    }
+
+    @Override
+    protected boolean initIsRDD() {
+        // TODO: figure out how to set isRDD for varRefIter.
```

Comment:
This is related to a discussion we had previously on dynamic function calls, which have the same issue: isRDD is actually context dependent, although for many expressions it is not (which is why, for now, we did not pass the context here). Could you open an issue to address it separately?

Reply:
Thanks a lot! I will open the issue. I believe this issue will be the last mile in fully supporting DF&RDD enabled initial let clauses.